### PR TITLE
Folder for tar file in Unnattended installer

### PR DESCRIPTION
**Pull request by c-bordon**
|Related issue|
|---|
|https://github.com/wazuh/wazuh-packages/issues/1300|

## Description

A subdirectory is created inside the tar named wazuh-install-files.
it is tested on Centos 7 and Ubuntu 20.04 on both systems it works ok.

### Centos 7

```
09/03/2022 20:33:56 INFO: --- Wazuh dashboard ---
09/03/2022 20:33:56 INFO: Starting Wazuh dashboard installation.
09/03/2022 20:35:17 INFO: Wazuh dashboard installation finished.
09/03/2022 20:35:17 INFO: Wazuh dashboard post-install configuration finished.
09/03/2022 20:35:17 INFO: Starting service wazuh-dashboard.
09/03/2022 20:35:17 INFO: Wazuh-dashboard service started.
09/03/2022 20:35:33 INFO: Starting Wazuh dashboard.
09/03/2022 20:35:45 INFO: Wazuh dashboard started.
09/03/2022 20:35:45 INFO: --- Summary ---
09/03/2022 20:35:45 INFO: You can access the web interface https://<wazuh-dashboard-ip>.
    User: admin
    Password: un7SByIujIPtZaftRdA6loeFHWrjHSE2
09/03/2022 20:35:45 INFO: Installation finished.
09/03/2022 20:35:45 INFO: The certificates and passwords used are stored in /root/wazuh-packages/unattended_installer/wazuh-install-files.tar.

```
Screenshot_20220309_174020

```
[root@centos7-1 unattended_installer]# curl -X GET -k -u admin:un7SByIujIPtZaftRdA6loeFHWrjHSE2 https://localhost:9200/_cat/health
1646858508 20:41:48 wazuh-cluster yellow 1 1 true 9 9 0 0 1 0 - 90.0%

```

### Ubuntu 20.04
```
09/03/2022 20:36:49 INFO: --- Wazuh dashboard ---
09/03/2022 20:36:49 INFO: Starting Wazuh dashboard installation.
09/03/2022 20:37:30 INFO: Wazuh dashboard installation finished.
09/03/2022 20:37:30 INFO: Wazuh dashboard post-install configuration finished.
09/03/2022 20:37:30 INFO: Starting service wazuh-dashboard.
09/03/2022 20:37:30 INFO: Wazuh-dashboard service started.
09/03/2022 20:37:44 INFO: Starting Wazuh dashboard.
09/03/2022 20:37:56 INFO: Wazuh dashboard started.
09/03/2022 20:37:56 INFO: --- Summary ---
09/03/2022 20:37:56 INFO: You can access the web interface https://<wazuh-dashboard-ip>.
    User: admin
    Password: xBtf9QiWidAf6h6FgSsw2v5bxF1cesLt
09/03/2022 20:37:56 INFO: Installation finished.
09/03/2022 20:37:56 INFO: The certificates and passwords used are stored in /root/wazuh-packages/unattended_installer/wazuh-install-files.tar.
root@ubuntu20:~/wazuh-packages/unattended_installer# 

```
Screenshot_20220309_174316

```
root@ubuntu20:~/wazuh-packages/unattended_installer# curl -X GET -k -u admin:xBtf9QiWidAf6h6FgSsw2v5bxF1cesLt https://localhost:9200/_cat/health
1646858632 20:43:52 wazuh-cluster yellow 1 1 true 7 7 0 0 1 0 - 87.5%

```

```
[root@centos7-1 unattended_installer]# tar tf wazuh-install-files.tar 
wazuh-install-files/
wazuh-install-files/root-ca.key
wazuh-install-files/root-ca.pem
wazuh-install-files/admin-key.pem
wazuh-install-files/admin.pem
wazuh-install-files/indexer-key.pem
wazuh-install-files/indexer.pem
wazuh-install-files/server-key.pem
wazuh-install-files/server.pem
wazuh-install-files/dashboard-key.pem
wazuh-install-files/dashboard.pem
wazuh-install-files/password_file.yml
wazuh-install-files/config.yml
```

## Tests

- Build the package in any supported platform
  - [ ] Linux
  - [ ] Windows
  - [ ] macOS
  - [ ] Solaris
  - [ ] AIX
  - [ ] HP-UX
- [x] Package installation
- [ ] Package upgrade
- [ ] Package downgrade
- [ ] Package remove
- [x] Package install/remove/install
- [ ] Change added to CHANGELOG.md

- Tests for Linux RPM
  - [ ] Build the package for x86_64
  - [ ] Build the package for i386
  - [ ] Build the package for armhf
  - [ ] Build the package for aarch64
  - [ ] `%files` section is correctly updated if necessary
- Tests for Linux deb
  - [ ] Build the package for x86_64
  - [ ] Build the package for i386
  - [ ] Build the package for armhf
  - [ ] Build the package for aarch64
  - [x] Package install/remove/install
  - [x] Package install/purge/install
  - [ ] Check file permissions after installing the package
- Tests for macOS
  - [ ] Test the package from macOS Sierra to Mojave
- Tests for Solaris
  - [ ] Test the package on Solaris 10
  - [ ] Test the package on Solaris 11
  - [ ] Check file permissions on Solaris 11 template
- Tests for IBM AIX
  - [ ] `%files` section is correctly updated if necessary
  - [ ] Check the changes from IBM AIX 5 to 7

### CODE REVIEW

**Review by rauldpm — CHANGES_REQUESTED** [review]
GJ, please review changes

File: unattended_installer/install_functions/dashboard.sh [hunk 1]
```diff
@@ -49,9 +49,10 @@ function dashboard_copyCertificates() {
 
         name=${dashboard_node_names[pos]}
 
-        eval "tar -xf ${tar_file} -C ${dashboard_cert_path} ./${name}.pem  && mv ${dashboard_cert_path}${name}.pem ${dashboard_cert_path}dashboard.pem ${debug}"
-        eval "tar -xf ${tar_file} -C ${dashboard_cert_path} ./${name}-key.pem  && mv ${dashboard_cert_path}${name}-key.pem ${dashboard_cert_path}dashboard-key.pem ${debug}"
-        eval "tar -xf ${tar_file} -C ${dashboard_cert_path} ./root-ca.pem ${debug}"
+        eval "tar -xf ${tar_file} -C ${dashboard_cert_path} wazuh-install-files/${name}.pem  && mv ${dashboard_cert_path}wazuh-install-files/${name}.pem ${dashboard_cert_path}dashboard.pem ${debug}"
+        eval "tar -xf ${tar_file} -C ${dashboard_cert_path} wazuh-install-files/${name}-key.pem  && mv ${dashboard_cert_path}wazuh-install-files/${name}-key.pem ${dashboard_cert_path}dashboard-key.pem ${debug}"
+        eval "tar -xf ${tar_file} -C ${dashboard_cert_path} wazuh-install-files/root-ca.pem && mv ${dashboard_cert_path}wazuh-install-files/root-ca.pem ${dashboard_cert_path}root-ca.pem ${debug}"
+        eval "rm -rf ${dashboard_cert_path}wazuh-install-files/"
```
[the remark this answers is not in the document]
Comment (c-bordon) [reply unattended_installer/install_functions/dashboard.sh]: Done: https://github.com/wazuh/wazuh-packages/pull/1319/commits/311482cb64678289950e3d3617caea5448b3ba0a

File: unattended_installer/install_functions/indexer.sh [hunk 2]
```diff
@@ -80,12 +80,13 @@ function indexer_copyCertificates() {
...
-        eval "tar -xf ${tar_file} -C ${indexer_cert_path} ./admin-key.pem  ${debug}"
-        eval "chown wazuh-indexer:wazuh-indexer ${indexer_cert_path}*"
+        eval "tar -xf ${tar_file} -C ${indexer_cert_path} wazuh-install-files/${name}.pem  && mv ${indexer_cert_path}wazuh-install-files/${name}.pem ${indexer_cert_path}indexer.pem ${debug}"
+        eval "tar -xf ${tar_file} -C ${indexer_cert_path} wazuh-install-files/${name}-key.pem  && mv ${indexer_cert_path}wazuh-install-files/${name}-key.pem ${indexer_cert_path}indexer-key.pem ${debug}"
+        eval "tar -xf ${tar_file} -C ${indexer_cert_path} wazuh-install-files/root-ca.pem && mv ${indexer_cert_path}wazuh-install-files/root-ca.pem ${indexer_cert_path} ${debug}"
+        eval "tar -xf ${tar_file} -C ${indexer_cert_path} wazuh-install-files/admin.pem && mv ${indexer_cert_path}wazuh-install-files/admin.pem ${indexer_cert_path} ${debug}"
+        eval "tar -xf ${tar_file} -C ${indexer_cert_path} wazuh-install-files/admin-key.pem && mv ${indexer_cert_path}wazuh-install-files/admin-key.pem ${indexer_cert_path} ${debug}"
+        eval "rm -rf ${indexer_cert_path}wazuh-install-files/"
```
[the remark this answers is not in the document]
Comment (c-bordon) [reply unattended_installer/install_functions/indexer.sh]: Done: https://github.com/wazuh/wazuh-packages/pull/1319/commits/311482cb64678289950e3d3617caea5448b3ba0a

**Review by alberpilot — CHANGES_REQUESTED** [review]
Modify the unattended install test accordingly.

**Comment by c-bordon** [comment]
Problems solved with distributed test

Distributed: https://devel.ci.wazuh.info/view/Tests/job/Test_unattended_distributed/360/console

AIO:
Centos 7: https://devel.ci.wazuh.info/view/Tests/job/Test_unattended/475/console
Ubuntu Bionic: https://devel.ci.wazuh.info/view/Tests/job/Test_unattended/474/console

**Review by rauldpm — APPROVED** [review]
LGTM

**Review by alberpilot — APPROVED** [review]

**Pull request merged**